tests/services/sessions/test_call_context.py

```diff
@@ -0,0 +1,104 @@
+import asyncio
+import time
+
+from jupyter_server.auth.utils import get_anonymous_username
+from jupyter_server.base.handlers import JupyterHandler
+from jupyter_server.services.kernels.kernelmanager import AsyncMappingKernelManager
+from jupyter_server.services.sessions.call_context import CallContext
+
+
+async def test_jupyter_handler_contextvar(jp_fetch, monkeypatch):
+    # Create some mock kernel Ids
+    kernel1 = "x-x-x-x-x"
+    kernel2 = "y-y-y-y-y"
+
+    # We'll use this dictionary to track the current user within each request.
+    context_tracker = {
+        kernel1: {"started": "no user yet", "ended": "still no user", "user": None},
+        kernel2: {"started": "no user yet", "ended": "still no user", "user": None},
+    }
+
+    # Monkeypatch the get_current_user method in Tornado's
+    # request handler to return a random user name for
+    # each request
+    async def get_current_user(self):
+        return get_anonymous_username()
+
+    monkeypatch.setattr(JupyterHandler, "get_current_user", get_current_user)
+
+    # Monkeypatch the kernel_model method to show that
+    # the current context variable is truly local and
+    # not contaminated by other asynchronous parallel requests.
+    def kernel_model(self, kernel_id):
```

Comment:
I agree. This is why I added `test_multi_context_operations`.  Do you have any ideas on another method to monkeypatch?

Reply:
Just pushed an updated 😎

Reply:
Ok - thanks.  I had just found that we _could_ convert the monkey-patched `kernel_model` to async because we wrap the call in `ensure_async()` from the handler.  This would prevent having to start a kernel.

(I think we should convert `time.sleep` to `asyncio.sleep`.

Here's what I had tested just prior to your push:
```python
    # Monkeypatch the kernel_model method to show that
    # the current context variable is truly local and
    # not contaminated by other asynchronous parallel requests.
    # Note that even though the current implementation of `kernel_model()`
    # is synchronous, we can convert this into an async method because the
    # kernel handler wraps the call to `kernel_model()` in `ensure_async()`.
    async def kernel_model(self, kernel_id):
        # Get the Jupyter Handler from the current context.
        current: JupyterHandler = CallContext.get(CallContext.JUPYTER_HANDLER)
        # Get the current user
        context_tracker[kernel_id]["user"] = current.current_user
        context_tracker[kernel_id]["started"] = current.current_user
        await asyncio.sleep(1.0)
        # Track the current user a few seconds later. We'll
        # verify that this user was unaffected by other parallel
        # requests.
        context_tracker[kernel_id]["ended"] = current.current_user
        return {"id": kernel_id, "name": "blah"}
```

Reply:
Ah nice. Let's do that. I'll update the test back to this.